pyproject.toml

```diff
@@ -26,15 +26,38 @@ profile = "black"
 known_third_party = ["distutils"]
 force_to_top = ["setuptools"]
 
+
 [tool.mypy]
 files = ["skbuild"]
 python_version = "3.6"
 warn_unused_configs = true
 check_untyped_defs = true
 show_error_codes = true
 enable_error_code = ["ignore-without-code", "redundant-expr", "truthy-bool"]
-warn_unreachable = true
+strict_equality = true
+strict_concatenate = true
+disallow_untyped_decorators = true
+no_implicit_optional = true
+warn_redundant_casts = true
+warn_unused_ignores = true
+no_implicit_reexport = true
+disallow_any_generics = true
+disallow_incomplete_defs = true
+
+# disallow_subclassing_any = true 4
+# warn_return_any = true 5
+# disallow_untyped_calls = true 39
+# disallow_untyped_defs = true 61
```

Comment:
These are the remaining failures that need addressing before this is fully strictly typed. Started a bit on `disallow_untyped_defs` but it looks like the comments lie a bit, so will take a bit of work. Best to get this in as a strict improvement, I think, then work on these separately.